tests/test_non_uniform_cmfd_linear_source/test_non_uniform_cmfd_linear_source.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+
+import os
+import sys
+sys.path.insert(0, os.pardir)
+sys.path.insert(0, os.path.join(os.pardir, 'openmoc'))
+from testing_harness import TestHarness
+from input_set import PwrAssemblyInput
+import openmoc
+
+
+class CmfdPwrAssemblyTestHarness(TestHarness):
+    """An eigenvalue calculation for a 17x17 lattice with 7-group C5G7
+    cross section data."""
+
+    def __init__(self):
+        super(CmfdPwrAssemblyTestHarness, self).__init__()
+        self.input_set = PwrAssemblyInput()
+
+    def _create_geometry(self):
+        """Initialize CMFD and add it to the Geometry."""
+
+        super(CmfdPwrAssemblyTestHarness, self)._create_geometry()
+
+        # Initialize CMFD
+        cmfd = openmoc.Cmfd()
+        cmfd.setCMFDRelaxationFactor(0.7)
+        #cmfd.setLatticeStructure(17,17)
+        cmfd.setWidths([[1.26]*17, [1.26]*17])
```

Comment:
this lattice is uniform, but built with a non-uniform function. It's probably more important to test a truly non-uniform case

Reply:
test_non_uniform_lattice_geometry.py will do this

Reply:
Do you think we should keep that test then ?